cli/flox-rust-sdk/src/providers/services.rs

```diff
@@ -785,4 +829,110 @@ mod tests {
+                    vars: None,
+                }),
+            ]
+            .into(),
+        });
+        stop_services(instance.socket(), &vec!["proc_stopped"]).unwrap();
+        let mut states = ProcessStates::read(instance.socket()).unwrap();
+
+        // Use predictable PIDs and exercise right alignment of the column.
+        let mut next_pid = 10;
+        for proc in &mut states.0 {
+            proc.pid = next_pid;
+            next_pid *= 10;
+        }
+
+        // Table output.
+        assert_eq!(format!("{states}"), indoc! {"
+            NAME                 STATUS          PID
+            aaa_first            Completed        10
+            proc_completed       Completed       100
+            proc_running         Running        1000
+            proc_stopped         Completed     10000
+            zzz_last             Completed    100000
+        "});
+
+        // JSON lines output.
+        assert_eq!(format!("{:#}", states), indoc! {r#"
+            {"name":"aaa_first","status":"Completed","pid":10}
+            {"name":"proc_completed","status":"Completed","pid":100}
+            {"name":"proc_running","status":"Running","pid":1000}
+            {"name":"proc_stopped","status":"Completed","pid":10000}
+            {"name":"zzz_last","status":"Completed","pid":100000}
+        "#});
+    }
+
+    #[test]
+    fn test_processstates_read_names() {
+        let instance = TestProcessComposeInstance::start(&ProcessComposeConfig {
+            processes: [
+                ("aaa".to_string(), ProcessConfig {
+                    command: "true".to_string(),
+                    vars: None,
+                }),
+                ("bbb".to_string(), ProcessConfig {
+                    command: "true".to_string(),
+                    vars: None,
+                }),
+                ("ccc".to_string(), ProcessConfig {
+                    command: "true".to_string(),
+                    vars: None,
+                }),
+                ("ddd".to_string(), ProcessConfig {
+                    command: "true".to_string(),
+                    vars: None,
+                }),
+            ]
+            .into(),
+        });
+        let mut states = ProcessStates::read_names(instance.socket(), vec![
+            "aaa".to_string(),
+            "ccc".to_string(),
+            "unknown".to_string(),
+        ])
+        .unwrap();
+
+        // Use a predictable PID.
+        for proc in &mut states.0 {
+            proc.pid = 12345;
+        }
+
+        // Table output
+        assert_eq!(format!("{states}"), indoc! {"
+            NAME                 STATUS          PID
+            aaa                  Completed     12345
+            ccc                  Completed     12345
+        "});
+
+        // JSON lines output.
+        assert_eq!(format!("{:#}", states), indoc! {r#"
+            {"name":"aaa","status":"Completed","pid":12345}
+            {"name":"ccc","status":"Completed","pid":12345}
+        "#});
+    }
```

Comment:
blocking: add a test for overflow formatting

Reply:
Can you explain more? Do you mean fields overflowing the fixed padding of the table output?

Reply:
Yeah I meant for when the contents of the column are wider than the column

Reply:
Cool. The refactor included a test for a longer name and a subsequent commit now resizes the first column.